### PR TITLE
feat(base-images): build base image during setup process

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

**Comment by cooperaj** [comment]
Pulling and building now...

**Comment by cooperaj** [comment]
Think something may be broken.

### Steps to reproduce:

1. `npm run build-image`
2. Run it
3. Open localhost 6500
4. Register admin account
5. Base image is created 👍 
6. Create builder image called 'abstruse' - no changes to pre-filled fields
7. `node tests/dev-scripts/pull-request.js`
8. Build fails with missing NVM.

File: src/app/components/app-images/app-images.component.html [hunk 1]
```diff
@@ -74,14 +98,45 @@
...
             <div class="columns is-multiline">
               <div class="column is-12">
+                <div class="content-box approve" *ngIf="approve">
+                  <div class="columns is-multiline">
+                    <div class="column is-12">
+                      <h1>Warning:</h1>
+                    </div>
+                    <div class="column is-12">
+                        <div class="column is-12">
+                          Your docker file contains commands that can corrupt Abstruse to work properly ({{dangerousCommands}}).
```
Comment (cooperaj) [inline src/app/components/app-images/app-images.component.html]: Grammar change:
"Your docker file contains commands that can cause Abstruse to not work correctly ({{dangerousCommands}}).

**Comment by cooperaj** [comment]
### Comments

- I think the UI on the images page needs a bit of work.

**Comment by Izak88** [comment]
You're right, something is broken. This procedure worked for me the first time I try, but I guess I wreck something... 
Will fix that, tnx.

File: src/app/components/app-images/app-images.component.html (hunk 1 above)
Comment (Izak88) [reply src/app/components/app-images/app-images.component.html]: tnx, will fix that...

File: src/api/image-builder.ts [hunk 2]
```diff
@@ -21,9 +22,16 @@ export interface ImageBuildOutput {
 export const imageBuilder: Subject<ImageBuildOutput> = new Subject();
 export const imageBuilderObs = imageBuilder.share();
 
+export function buildAbstruseBaseImage(): Promise<any> {
+  return Promise.resolve().then(() => {
```
Comment (jkuri) [inline src/api/image-builder.ts]: you can do
```ts
return buildDockerImage(defaultBaseImage);
```

File: src/api/server-routes.ts [hunk 3]
```diff
@@ -770,6 +770,11 @@ export function imagesRoutes(): express.Router {
       .catch(err => res.status(200).json({ status: false }));
   });
 
+  router.post('/build-base', (req: express.Request, res: express.Response) => {
+    buildAbstruseBaseImage()
+      .then(() => res.status(200).json({ data: true }));
```
Comment (jkuri) [inline src/api/server-routes.ts]: I think it would be smart here to `.catch`.

File: docs/IMAGES.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# Abstruse CI Images
+
+**For running builds Docker Image is required.**
+
+During the process of Abstruse installation, a new Docker Image with name `abstruse_builder` is built. This Image provides Abstruse
+with all of the necessary data to run builds.
+
+Each config file `.abstruse.yml` in your repositor must start with line `image: "name_of_abstruse_image"`, where `"name_of_abstruse_image"` represents
```
Comment (cooperaj) [inline docs/IMAGES.md]: TYPO: "repository"

Comment (cooperaj) [reply docs/IMAGES.md]: In fact, there are a few. I'll wait till the PR goes in and do a new one with corrections :)

Comment (jkuri) [reply docs/IMAGES.md]: thanks for watching our back Adam, its nice to have a native English speaker in the team.

Comment (Izak88) [reply docs/IMAGES.md]: hehe tnx, we're not english native speaker and it shows quickly when you start reading our documentation :).

File: src/api/image-builder.ts (hunk 2 above)
Comment (Izak88) [reply src/api/image-builder.ts]: done

File: src/api/server-routes.ts (hunk 3 above)
Comment (Izak88) [reply src/api/server-routes.ts]: done

**Comment by jkuri** [comment]
we merge this too soon, cannot be on master on that shape and I guess nobody will fix it will Monday.

**Pull request merged**